shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Ppup.hs

```diff
@@ -70,6 +72,8 @@ data PpupPredicateFailure era
       !VotingPeriod -- voting period within the epoch
   | PVCannotFollowPPUP
       !ProtVer -- the first bad protocol version
+  | DNotAHundredth
+      !Rational -- the first bad value of d
```

Comment:
FYI, I think this type is nested in the ledger errors type that will be sent to clients, who will fail to decode it.

Reply:
you're right. I'm not sure what the best thing to do is. We can forgo the check (which is not a huge deal, the new overlay style actually copes much better with wonky values of `d`). Or we could return a misleading error. :thinking:

Reply:
I also don't know what the best thing to do is. Adding is less harmful than removing or modifying, though. @dcoutts Thoughts on this?
Related: https://github.com/input-output-hk/ouroboros-network/issues/2488

Thought: are these errors that we sent over the network actually inspected? If not, just using `Text` (or using `Text` embedded in a data type with extra metadata) would be an easy way out, after the initial switch.

Reply:
> Thought: are these errors that we sent over the network actually inspected? If not, just using `Text` (or using `Text` embedded in a data type with extra metadata) would be an easy way out, after the initial switch. The conversion to `Text` would happen in consensus, of course, the ledger would still use structured types.

I do not know if they are inspected anywhere. Maybe @dcoutts knows? I like this plan if we can get away with it, though.

Reply:
I've removed the validation on `d` for now. The new schedule is actually much better with small values of `d` anyway.